src/reanimated2/shareables.ts

```diff
@@ -241,14 +241,28 @@ Offending code was: \`${getWorkletCode(value)}\``);
         shareableMappingCache.set(value, inaccessibleObject);
         return inaccessibleObject;
       }
-      if (__DEV__) {
-        // we freeze objects that are transformed to shareable. This should help
-        // detect issues when someone modifies data after it's been converted to
-        // shareable. Meaning that they may be doing a faulty assumption in their
+      if (__DEV__ && !isHostObject(value) && isPlainJSObject(value)) {
+        // We freeze objects that are transformed to Shareable with a meaningful warning.
```

Comment:
```suggestion
        // We freeze objects that are transformed to a shareable with a meaningful warning.
```

Reply:
I'd like to keep capital Shareable to give a distinction that it's a specific concept

Reply:
I won't fight about this one but I still think we need to stick with lower-case terms (i.e. shared value, mutable, shareable).